### PR TITLE
[improvement](file cache) Try to read from remote storage when opening segment with CachedRemoteFileReader

### PR DESCRIPTION
**Pull request by gavinchou**
We may encounter the error "Bad segment" file in some rare cases where the file cache may not hold the correct segment files. We should read the remote original segment files to increase robustness.

### CODE REVIEW

**Comment by gavinchou** [comment]
run buildall

**Review by platoneko — DISMISSED** [review]

**Comment by gavinchou** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 1]
```diff
@@ -77,16 +79,45 @@ namespace doris::segment_v2 {
...
+    auto st = segment->_open();
+    TEST_INJECTION_POINT_CALLBACK("Segment::open:corruption", &st);
+    if (st.is<ErrorCode::CORRUPTION>() &&
+        reader_options.cache_type == io::FileCachePolicy::FILE_BLOCK_CACHE) {
+        LOG(WARNING) << "bad segment file may be read from file cache, try to read remote source "
+                        "file directly, file path: "
+                     << path << " cache_key: " << file_cache_key_str(path);
+        auto file_key = file_cache_key_from_path(path);
+        auto* file_cache = io::FileCacheFactory::instance()->get_by_path(file_key);
+        file_cache->remove_if_cached(file_key);
```
Comment (freemandealer) [inline be/src/olap/rowset/segment_v2/segment.cpp:109]: the corruption file may not be deleted right  away if the cell is not releasable

**Comment by dataroaring** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Hastyshell — APPROVED** [review]
LGTM

**Pull request merged**